soccer/main.cpp

```diff
@@ -114,6 +120,27 @@ int main(int argc, char* argv[]) {
             playbookFile = argv[++i];
         } else if (strcmp(var, "-noref") == 0) {
             noref = true;
+        } else if (strcmp(var, "-defend") == 0) {
+            if (i + 1 >= argc) {
+                printf("Field half not specified after -defend\n");
+                usage(argv[0]);
+            }
+            i++;
+            if (strcmp(argv[i], "plus") == 0) {
+                defendPlus = true;
+            }
+        } else if (strcmp(var, "-vision") == 0) {
+            if (i + 1 >= argc) {
+                printf("No vision channel specified after -vision\n");
+                usage(argv[0]);
+            }
+            i++;
+            // USE STRCMP
+            if (strcmp(argv[i], "1") == 0) {
+                visionChannel = 1;
+            } else if (strcmp(argv[i], "2") == 0) {
+                visionChannel = 2;
+            }
```

Comment:
I think that you want to check if full is present here (and if it isn't we should error out, and not fail silently and assume full).